### PR TITLE
Humidity Sensor and change speed error fix for zhimi.humidifier.ca1

### PR DESCRIPTION
**Pull request by IldarMurzenkov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
-      if (value > 0) {
-        [ result ] = await this.device.call('set_mode', [speedToMode[value]])
-      } else {
-        if (this.model === 'ca1') {
-          [ result ] = await this.device.call('set_mode', ['auto'])
-        } else {
-          [ result ] = await this.device.call('set_power', ['off'])
+      const [ power ] = await this.device.call('get_prop', ['power'])
+      if (value > 0){
+        if (poewer == 'off'){
```
Comment (russtone) [inline index.js]:
```suggestion
        if (power === 'off') {
```

File: index.js [hunk 2]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
+          await this.device.call('set_power', ['on'])
+        }
+        if (value < 4){
```
Comment (russtone) [inline index.js]:
```suggestion
        if (value < 4) {
```

File: index.js [hunk 3]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
+         [ result ] = await this.device.call('set_mode', [speedToMode[value]])
```
Comment (russtone) [inline index.js]:
```suggestion
          [ result ] = await this.device.call('set_mode', [speedToMode[value]])
```

File: index.js [hunk 4]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
         }
+        if(value === 4 && this.model === 'ca1'){
```
Comment (russtone) [inline index.js]:
```suggestion
        if (value === 4 && this.model === 'ca1') {
```

File: index.js [hunk 5]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
+         [ result ] = await this.device.call('set_mode', ['auto'])
```
Comment (russtone) [inline index.js]:
```suggestion
          [ result ] = await this.device.call('set_mode', ['auto'])
```

File: index.js [hunk 6]
```diff
@@ -235,15 +248,19 @@ class MiHumidifier {
...
-
-      if (value > 0) {
-        [ result ] = await this.device.call('set_mode', [speedToMode[value]])
-      } else {
-        if (this.model === 'ca1') {
-          [ result ] = await this.device.call('set_mode', ['auto'])
-        } else {
-          [ result ] = await this.device.call('set_power', ['off'])
+      const [ power ] = await this.device.call('get_prop', ['power'])
+      if (value > 0){
```
Comment (russtone) [inline index.js]:
```suggestion
      if (value > 0) {
```

**Comment by IldarMurzenkov** [comment]
Hello, thank you for reply. I have fixed your comments.

**Pull request merged**